org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/MakeStaticRefactoring.java

```diff
@@ -0,0 +1,249 @@
+	}
+
+	/**
+	 * Constructs a new {@code MakeStaticRefactoring} object. This constructor is called when
+	 * performing the refactoring on a method in the outline menu.
+	 *
+	 * @param method The target method the refactoring should be performed on.
+	 */
+	public MakeStaticRefactoring(IMethod method) {
+		fContextCalculator= new ContextCalculator(method);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @return The name of the refactoring operation.
+	 */
+	@Override
+	public String getName() {
+		return RefactoringCoreMessages.MakeStaticRefactoring_name;
+	}
+
+	/**
+	 * Checks the initial conditions required for the refactoring process.
+	 *
+	 * @param pm The progress monitor to report the progress of the operation.
+	 * @return The refactoring status indicating the result of the initial conditions check.
+	 */
+	@Override
+	public RefactoringStatus checkInitialConditions(IProgressMonitor pm) {
+		fStatus= new RefactoringStatus();
+
+		SelectionInputType selectionInputType= fContextCalculator.getSelectionInputType();
+
+		if (selectionInputType == SelectionInputType.TEXT_SELECTION) {
+			fStatus.merge(InitialConditionsChecker.checkTextSelectionStart(fContextCalculator.getSelectionEditorText()));
+			fStatus.merge(InitialConditionsChecker.checkValidICompilationUnit(fContextCalculator.getSelectionICompilationUnit()));
+			if (fStatus.hasError()) {
+				return fStatus;
+			}
+
+			fStatus.merge(InitialConditionsChecker.checkASTNodeIsValidMethod(fContextCalculator.getOrComputeSelectionASTNode()));
+			if (fStatus.hasError()) {
+				return fStatus;
+			}
+		}
+
+		fStatus.merge(InitialConditionsChecker.checkSourceAvailable(fContextCalculator.getOrComputeTargetIMethod()));
+		if (fStatus.hasError()) {
+			return fStatus;
+		}
+
+		fStatus.merge(InitialConditionsChecker.checkIMethodIsValid(fContextCalculator.getOrComputeTargetIMethod()));
+		if (fStatus.hasError()) {
+			return fStatus;
+		}
+
+		fStatus.merge(InitialConditionsChecker.checkValidICompilationUnit(fContextCalculator.getOrComputeTargetICompilationUnit()));
+		if (fStatus.hasError()) {
+			return fStatus;
+		}
+
+		fStatus.merge(InitialConditionsChecker.checkMethodIsNotConstructor(fContextCalculator.getOrComputeTargetIMethod()));
+		if (fStatus.hasError()) {
+			return fStatus;
+		}
+
+		fStatus.merge(InitialConditionsChecker.checkMethodNotInLocalOrAnonymousClass(fContextCalculator.getOrComputeTargetIMethod()));
+		if (fStatus.hasError()) {
+			return fStatus;
+		}
+
+		fStatus.merge(InitialConditionsChecker.checkMethodNotStatic(fContextCalculator.getOrComputeTargetIMethod()));
+		if (fStatus.hasError()) {
+			return fStatus;
+		}
+
+		fStatus.merge(InitialConditionsChecker.checkMethodNotOverridden(fContextCalculator.getOrComputeTargetIMethod()));
+		if (fStatus.hasError()) {
+			return fStatus;
+		}
+
+
+		fTargetMethod= fContextCalculator.getOrComputeTargetIMethod(); //TODO remove those which dont have to necessarily be fields
```

Comment:
is that todo still valid?

Reply:
Nope and this line is also deleted already. Change is in 6061d068c3.